// form-table-rubber/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports["default"] = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _button = _interopRequireDefault(require("antd/lib/button"));
var _form = _interopRequireDefault(require("antd/lib/form"));
var _input = _interopRequireDefault(require("antd/lib/input"));
var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));
var _react = _interopRequireWildcard(require("react"));
var React = _react;
var _excluded = ["api", "type"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
// import type { FormProps } from 'antd';

// type FieldType = {
//   username?: string;
//   password?: string;
// };

// const onFinish: FormProps<FieldType>['onFinish'] = (values:any) => {
//   console.log('Success:', values);
// };

// const onFinishFailed: FormProps<FieldType>['onFinishFailed'] = (errorInfo:any) => {
//   console.log('Failed:', errorInfo);
// };

var ExampleComponent = function ExampleComponent(props, ref) {
  var api = props.api,
    _props$type = props.type,
    type = _props$type === void 0 ? 'tableForm' : _props$type,
    others = (0, _objectWithoutPropertiesLoose2["default"])(props, _excluded);
  var _useState = (0, _react.useState)(''),
    name = _useState[0],
    setName = _useState[1];
  var _useState2 = (0, _react.useState)(''),
    id = _useState2[0],
    setId = _useState2[1];
  return /*#__PURE__*/React.createElement("div", {
    className: "lowcode-spa-queryform",
    ref: ref
  }, /*#__PURE__*/React.createElement(_form["default"], (0, _extends2["default"])({
    name: "basic",
    labelCol: {
      span: 8
    },
    wrapperCol: {
      span: 16
    },
    style: {
      marginTop: 20,
      maxWidth: 500,
      display: 'flex',
      flexDirection: 'row'
    }
    // onFinish={onFinish}
    // onFinishFailed={onFinishFailed}
    ,
    autoComplete: "off",
    ref: ref
  }, others), /*#__PURE__*/React.createElement(_form["default"].Item, {
    label: "Name",
    name: "name"
  }, /*#__PURE__*/React.createElement(_input["default"], {
    onChange: function onChange(v) {
      setName(v.target.value);
    }
  })), /*#__PURE__*/React.createElement(_form["default"].Item, {
    label: "ID",
    name: "id"
  }, /*#__PURE__*/React.createElement(_input["default"], {
    onChange: function onChange(v) {
      setId(v.target.value);
    }
  })), type === 'tableForm' ? /*#__PURE__*/React.createElement(_form["default"].Item, {
    wrapperCol: {
      offset: 8,
      span: 16
    }
  }, /*#__PURE__*/React.createElement(_button["default"], {
    type: "primary",
    htmlType: "submit",
    onClick: function onClick() {
      console.log(name, id);
      var xhr = new XMLHttpRequest();
      xhr.open("GET", "https://" + (api || "proapi.azurewebsites.net/api/rule?token%20=%20123&current=1&pageSize=100") + "/" + name + id, true);
      xhr.onreadystatechange = function () {
        if (xhr.readyState === 4 && xhr.status === 200) {
          var res = JSON.parse(xhr.responseText);
          var data = res.data;
          console.log(data);
        }
      };
      xhr.send();
    }
  }, "Search")) : null));
};
var RefExampleComponent = /*#__PURE__*/(0, _react.forwardRef)(ExampleComponent);
RefExampleComponent.displayName = 'ExampleComponent';
var _default = exports["default"] = RefExampleComponent;
